File: docs/jupyter/notebooks/source/ans_pl05_generic_fun_type/p7/jl/triv/triv.jl
struct triv{T}
    x :: T
end

function triv_val(t :: triv{<:Any})
    t.x
end

function main()
    s = triv{Int64}(3)
    x = triv_val(s)
    t = triv{String}("hello")
    y = triv_val(t)
    println(x)
    println(y)
end

main()

File: jupyter/nb/source/pl05_generic_fun_type/p7/go/triv/triv.go
package main

type triv[T any] struct {
    x T
}

func triv_val[T any](t triv[T]) T {
    return t.x
}

func main() {
    s := triv[int]{3}
    x := triv_val(s)
    t := triv[string]{"hello"}
    y := triv_val(t)
    println(x)
    println(y)
}
